src/LanguageServer/Protocol/Workspaces/ILspFileBasedProgramWorkspaceProvider.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CommonLanguageServerProtocol.Framework;
+
+namespace Microsoft.CodeAnalysis.LanguageServer;
+
+internal interface ILspFileBasedProgramWorkspaceProvider : ILspService
```

Comment:
this is also a service?  but is used to make a service?  something doesn't make sense here with all these types and interfaces and relationships.